### PR TITLE
Eliminate more uses of SExpr.

### PR DESCRIPTION
**Pull request by abdoo8080**
This PR eliminates all remaining uses of `SExpr` outside of statistics.

### CODE REVIEW

File: src/smt/smt_engine.cpp [hunk 1]
```diff
@@ -1965,59 +1965,90 @@ void SmtEngine::setIsInternalSubsolver() { d_isInternalSubsolver = true; }
...
+
+  if (atom == "true")
+  {
+    return nm->mkConst<bool>(true);
+  }
+  else if (atom == "false")
+  {
+    return nm->mkConst<bool>(false);
+  }
+  else
```
Comment (ajreynol) [inline src/smt/smt_engine.cpp:2039]: Nit: you can delete this `else`

File: src/smt/command.cpp [hunk 2]
```diff
@@ -1641,15 +1640,14 @@ void GetAssignmentCommand::invoke(api::Solver* solver, SymbolManager* sm)
...
-    std::vector<SExpr> sexprs;
+    std::vector<api::Term> sexprs;
     for (size_t i = 0, nterms = terms.size(); i < nterms; i++)
     {
-      std::vector<SExpr> ss;
-      ss.emplace_back(SExpr::Keyword(names[i]));
-      ss.emplace_back(SExpr::Keyword(values[i].toString()));
-      sexprs.emplace_back(ss);
+      // Put the name of the expression inside of a variable name for printing
+      api::Term name = solver->mkVar(solver->getBooleanSort(), names[i]);
```
Comment (ajreynol) [inline src/smt/command.cpp:1647]: I assume its wrong to use a string constant here?  Why is that?

**Review by ajreynol — CHANGES_REQUESTED** [review]

File: src/smt/command.cpp (hunk 2 above)
Comment (abdoo8080) [reply src/smt/command.cpp:1648]: Modified the comment. Please take another look.

**Review by ajreynol — APPROVED** [review]
LGTM

**Pull request merged**